Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Tests/ComponentModeTestDoubles.h>
+#include <Tests/ComponentModeTestFixture.h>
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/API/EntityCompositionRequestBus.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/ViewportUi/ButtonGroup.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiSwitcher.h>
+
+namespace UnitTest
+{
+    using AzToolsFramework::ComponentModeFramework::AnotherPlaceholderEditorComponent;
+    using AzToolsFramework::ComponentModeFramework::PlaceholderEditorComponent;
+    using ComponentModeSwitcher = AzToolsFramework::ComponentModeFramework::ComponentModeSwitcher;
+
+    class ComponentModeSwitcherTestFixture : public ComponentModeTestFixture
```

Comment:
Suggestion: rather than deriving a whole new fixture, can we not type alias this fixture, a la `using ComponentModeSwitcherTestFixture = ComponentModeTestFixture;`? The only unintended consequences I could see with this approach would be either you or other users of `ComponentModeTestFixture` writing typed or parametrized tests, neither of which I can see. If in the future we need to further expand on `ComponentModeTestFixture` we can simply replace the type alias with a derived class of the same name.